source/module_hsolver/hsolver_pw.cpp

```diff
@@ -158,20 +158,30 @@ void HSolverPW<T, Device>::solve(hamilt::Hamilt<T, Device>* pHamilt,
 
     std::vector<Real> eigenvalues(pes->ekb.nr * pes->ekb.nc, 0);
 
-    if (this->is_first_scf == true)
+    if (this->is_first_scf)
     {
         is_occupied.resize(psi.get_nk() * psi.get_nbands(), true);
     }
     else
     {
-        for (size_t i = 0; i < psi.get_nk(); i++)
+        if (this->diago_full_acc)
         {
-            for (size_t j = 0; j < psi.get_nbands(); j++)
+            is_occupied.assign(is_occupied.size(), true);
+        }
+        else
+        {
+            for (int i = 0; i < psi.get_nk(); i++)
             {
-                if (pes->wg(i, j) < 1.0)
+                if (pes->klist->wk[i] > 0.0)
                 {
-                    is_occupied[i * psi.get_nbands() + j] = false;
-                }
+                    for (int j = 0; j < psi.get_nbands(); j++)
+                    {
+                        if (pes->wg(i, j) / pes->klist->wk[i] < 0.01)
```

Comment:
add a check in case wk[i]=0.0, which will cause segmentation fault.

Reply:
look line175: if (pes->klist->wk[i] > 0.0)